core/validation/specific_validations/towns_fund_round_four.py

```diff
@@ -350,16 +350,15 @@ def validate_funding_spent(workbook: dict[str, pd.DataFrame]) -> list["TownsFund
     project_ids = workbook["Project Details"]["Project ID"].tolist()
     funding_df = workbook["Funding"]
 
-    # pull funding spent for individual projects and programme wide total
-    funding_spent = {project: spend_per_project(funding_df, project) for project in project_ids}
-    funding_spent[programme_id] = sum(funding_spent.values())
+    # pull funding spent for individual projects into a DataFrame
+    funding_spent = pd.DataFrame({project: spend_per_project(funding_df, project) for project in project_ids})
+    funding_spent = funding_spent.transpose()  # swap rows and columns
 
     # TODO: create a single Failure instance for a single overspend error with a set of "locations" rather
     #   than a Failure for each cell
-    funding_spent_failures = []
     if fund_type == "HS":
         # check funding against programme wide funding allocated for Future High Street Fund submissions
-        if funding_spent[programme_id] > DefaultConfig.TF_FUNDING_ALLOCATED[programme_id]:
+        if funding_spent["Total"].sum() > DefaultConfig.TF_FUNDING_ALLOCATED["Total"][programme_id]:
```

Comment:
Can we wrap this `TF_Funding_Allocated` in some functions instead of directly interacting with it here? 

That way we can swap out the implementation of this more easily later if we change it for whatever reason - e.g. move it to being stored in DB...

Same for:
```
funding_spent[expense_type][idx] > DefaultConfig.TF_FUNDING_ALLOCATED[expense_type][idx]
```

Sounds like we're going to need to support this going fowards for new funds as well so could we make the interface fund agnostic like:
e.g.
```
get_programme_funding(fund: str, programme_id: str) -> int:
    pass
```

Reply:
I've done the inital steps for this. I think making this fund agnostic at this point may be complex and I'm consicious that I would like to get this into QA soon so that it can go out to users. SO we could save that for another ticket